src/coreclr/src/pal/src/loader/module.cpp

```diff
@@ -603,6 +615,12 @@ PAL_LoadLibraryDirect(
           lpLibFileName ? lpLibFileName : W16_NULLSTRING,
           lpLibFileName ? lpLibFileName : W16_NULLSTRING);
 
+    if (ShouldRedirectToCurrentLibrary(lpLibFileName))
```

Comment:
Nit: I do not think that the assert has much value. I would make this just `libraryNameOrPath == NULL`.

Reply:
I had doubts that we may be using NULL on this codepath for something else, perhaps transiently, especially when I saw failures on Linux arm/am64 related to "kernel32.dll not found". 
There was a thought - maybe NULL name _is_ involved in that scenario. However, having this assert I knew that it cannot be so.  It turned out something unrelated to this change and passed reliably after rebasing.

I guess at this point we know fairly that NULL is not used for anything else, so assert can be removed.